task_processing/interfaces/event.py

```diff
@@ -8,18 +8,36 @@
 
 
 class Event(PRecord):
-    timestamp = field(type=datetime.datetime)
+    type = field(type=str, initial=None)
```

Comment:
right, it is a problem that could be solved during serialization, but then you'd have to solve it multiple times in every place that reads our events, while having self-descriptive dict allows it to be interpreted in any environment with less hassle and side-steps the need to have special de/serialization procedures.

the reason i made these as separate types was to avoid having to provide timestamp every time we derive new event from existing one. we could replace those with `control_event` and `task_event` helper functions - would that be better?